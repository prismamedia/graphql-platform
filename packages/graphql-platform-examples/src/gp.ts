import { GraphQLPlatform } from '@prismamedia/graphql-platform';
import { config as Article } from './resources/article';

export const gp = new GraphQLPlatform({
  resources: {
    Article,
  },
});

console.debug(gp.schema);
