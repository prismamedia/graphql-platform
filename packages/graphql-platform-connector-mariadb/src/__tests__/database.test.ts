import sqlFormatter from 'sql-formatter';
import {
  CreateTableStatement,
  Database,
} from '../graphql-platform/connector/database';
import { config, MyGP } from './gp';

describe('Database', () => {
  let gp: MyGP;
  let database: Database;

  beforeAll(() => {
    gp = new MyGP(config);
    database = gp.getConnector().getDatabase();
  });

  it('creates a database', () => {
    expect(database).toBeInstanceOf(Database);
  });

  it('creates tables', () => {
    expect([...database.getTableSet()].map((table) => table.name))
      .toMatchInlineSnapshot(`
      Array [
        "categories",
        "users",
        "articles",
        "tags",
        "article_tags",
        "article_tag_comments",
        "article_urls",
        "article_url_metas",
      ]
    `);
  });

  it('print the "CREATE TABLE" statements', () => {
    expect(
      [...database.getTableSet()]
        .map((table) =>
          sqlFormatter.format(new CreateTableStatement(table).sql),
        )
        .join('\n\n'),
    ).toMatchSnapshot();
  });
});
