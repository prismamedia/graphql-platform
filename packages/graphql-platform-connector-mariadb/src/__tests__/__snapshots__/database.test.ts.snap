// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Database print the "CREATE TABLE" statements 1`] = `
"CREATE TABLE \`categories\` (
  \`intId\` INTEGER UNSIGNED NOT NULL AUTO_INCREMENT COMMENT 'The internal and private ID used to speed up some operations',
  \`id\` CHAR(36) NOT NULL COMMENT 'The public ID',
  \`createdAt\` TIMESTAMP(3) NOT NULL DEFAULT NOW(3) COMMENT 'The date, fixed, of the document\\\\'s creation',
  \`updatedAt\` TIMESTAMP(3) NOT NULL DEFAULT NOW(3) ON UPDATE NOW(3) COMMENT 'The date of the document\\\\'s last update',
  \`title\` VARCHAR(255) NOT NULL,
  \`slug\` VARCHAR(255) NOT NULL COMMENT 'The category\\\\'s slug',
  \`parentId\` CHAR(36) NULL,
  PRIMARY KEY (\`intId\`),
  UNIQUE \`unq_id\` (\`id\`),
  UNIQUE \`unq_parentId_slug\` (\`parentId\`, \`slug\`),
  INDEX (\`parentId\`),
  FOREIGN KEY (\`parentId\`) REFERENCES \`categories\` (\`id\`) ON UPDATE CASCADE ON DELETE RESTRICT
) ENGINE = 'InnoDB',
CHARSET = 'utf8mb4',
COLLATE = 'utf8mb4_unicode_520_ci';

CREATE TABLE \`users\` (
  \`intId\` INTEGER UNSIGNED NOT NULL AUTO_INCREMENT COMMENT 'The internal and private ID used to speed up some operations',
  \`id\` CHAR(36) NOT NULL COMMENT 'The public ID',
  \`createdAt\` TIMESTAMP(3) NOT NULL DEFAULT NOW(3) COMMENT 'The date, fixed, of the document\\\\'s creation',
  \`updatedAt\` TIMESTAMP(3) NOT NULL DEFAULT NOW(3) ON UPDATE NOW(3) COMMENT 'The date of the document\\\\'s last update',
  \`username\` VARCHAR(255) NOT NULL COMMENT 'The user\\\\'s username',
  PRIMARY KEY (\`intId\`),
  UNIQUE \`unq_id\` (\`id\`),
  UNIQUE \`unq_username\` (\`username\`)
) ENGINE = 'InnoDB',
CHARSET = 'utf8mb4',
COLLATE = 'utf8mb4_unicode_520_ci';

CREATE TABLE \`articles\` (
  \`intId\` INTEGER UNSIGNED NOT NULL AUTO_INCREMENT COMMENT 'The internal and private ID used to speed up some operations',
  \`id\` CHAR(36) NOT NULL COMMENT 'The public ID',
  \`createdAt\` TIMESTAMP(3) NOT NULL DEFAULT NOW(3) COMMENT 'The date, fixed, of the document\\\\'s creation',
  \`updatedAt\` TIMESTAMP(3) NOT NULL DEFAULT NOW(3) ON UPDATE NOW(3) COMMENT 'The date of the document\\\\'s last update',
  \`computedMaxIntId\` INTEGER NOT NULL COMMENT 'An \\\\\\"auto-increment\\\\\\" kind of field',
  \`publishedAt\` TIMESTAMP(3) NULL COMMENT 'The date of the document\\\\'s public release',
  \`format\` ENUM('RICH', 'VIDEO') NOT NULL,
  \`title\` VARCHAR(255) NOT NULL COMMENT 'The article\\\\'s title',
  \`slug\` VARCHAR(255) NOT NULL COMMENT 'The article\\\\'s slug',
  \`body\` VARCHAR(255) NULL COMMENT 'The article\\\\'s body',
  \`categoryParentId\` CHAR(36) NULL,
  \`categorySlug\` VARCHAR(255) NOT NULL,
  \`authorUsername\` VARCHAR(255) NOT NULL,
  \`moderatorIntId\` INTEGER UNSIGNED NULL,
  PRIMARY KEY (\`intId\`),
  UNIQUE \`unq_id\` (\`id\`),
  UNIQUE \`unq_computedMaxIntId\` (\`computedMaxIntId\`),
  UNIQUE \`unq_categoryParentId_categorySlug_slug\` (\`categoryParentId\`, \`categorySlug\`, \`slug\`),
  INDEX (\`categoryParentId\`, \`categorySlug\`),
  INDEX (\`authorUsername\`),
  INDEX (\`moderatorIntId\`),
  FOREIGN KEY (\`categoryParentId\`, \`categorySlug\`) REFERENCES \`categories\` (\`parentId\`, \`slug\`) ON UPDATE CASCADE ON DELETE RESTRICT,
  FOREIGN KEY (\`authorUsername\`) REFERENCES \`users\` (\`username\`) ON UPDATE CASCADE ON DELETE RESTRICT,
  FOREIGN KEY (\`moderatorIntId\`) REFERENCES \`users\` (\`intId\`) ON UPDATE CASCADE ON DELETE RESTRICT
) ENGINE = 'InnoDB',
CHARSET = 'utf8mb4',
COLLATE = 'utf8mb4_unicode_520_ci',
COMMENT = 'An article';

CREATE TABLE \`tags\` (
  \`intId\` INTEGER UNSIGNED NOT NULL AUTO_INCREMENT COMMENT 'The internal and private ID used to speed up some operations',
  \`id\` CHAR(36) NOT NULL COMMENT 'The public ID',
  \`createdAt\` TIMESTAMP(3) NOT NULL DEFAULT NOW(3) COMMENT 'The date, fixed, of the document\\\\'s creation',
  \`updatedAt\` TIMESTAMP(3) NOT NULL DEFAULT NOW(3) ON UPDATE NOW(3) COMMENT 'The date of the document\\\\'s last update',
  \`title\` VARCHAR(255) NOT NULL,
  \`slug\` VARCHAR(255) NOT NULL COMMENT 'The tag\\\\'s slug',
  PRIMARY KEY (\`intId\`),
  UNIQUE \`unq_id\` (\`id\`),
  UNIQUE \`unq_slug\` (\`slug\`)
) ENGINE = 'InnoDB',
CHARSET = 'utf8mb4',
COLLATE = 'utf8mb4_unicode_520_ci';

CREATE TABLE \`article_tags\` (
  \`articleIntId\` INTEGER UNSIGNED NOT NULL,
  \`tagIntId\` INTEGER UNSIGNED NOT NULL,
  \`createdAt\` TIMESTAMP(3) NOT NULL DEFAULT NOW(3) COMMENT 'The date, fixed, of the document\\\\'s creation',
  \`updatedAt\` TIMESTAMP(3) NOT NULL DEFAULT NOW(3) ON UPDATE NOW(3) COMMENT 'The date of the document\\\\'s last update',
  \`order\` INTEGER NOT NULL,
  PRIMARY KEY (\`articleIntId\`, \`tagIntId\`),
  UNIQUE \`article/order\` (\`articleIntId\`, \`order\`),
  INDEX (\`articleIntId\`),
  INDEX (\`tagIntId\`),
  FOREIGN KEY (\`articleIntId\`) REFERENCES \`articles\` (\`intId\`) ON UPDATE CASCADE ON DELETE RESTRICT,
  FOREIGN KEY (\`tagIntId\`) REFERENCES \`tags\` (\`intId\`) ON UPDATE CASCADE ON DELETE RESTRICT
) ENGINE = 'InnoDB',
CHARSET = 'utf8mb4',
COLLATE = 'utf8mb4_unicode_520_ci';

CREATE TABLE \`article_tag_comments\` (
  \`articleTagArticleIntId\` INTEGER UNSIGNED NOT NULL,
  \`articleTagTagIntId\` INTEGER UNSIGNED NOT NULL,
  \`createdAt\` TIMESTAMP(3) NOT NULL DEFAULT NOW(3) COMMENT 'The date, fixed, of the document\\\\'s creation',
  \`updatedAt\` TIMESTAMP(3) NOT NULL DEFAULT NOW(3) ON UPDATE NOW(3) COMMENT 'The date of the document\\\\'s last update',
  \`body\` VARCHAR(255) NULL,
  PRIMARY KEY (\`articleTagArticleIntId\`, \`articleTagTagIntId\`),
  INDEX (\`articleTagArticleIntId\`, \`articleTagTagIntId\`),
  FOREIGN KEY (\`articleTagArticleIntId\`, \`articleTagTagIntId\`) REFERENCES \`article_tags\` (\`articleIntId\`, \`tagIntId\`) ON UPDATE CASCADE ON DELETE RESTRICT
) ENGINE = 'InnoDB',
CHARSET = 'utf8mb4',
COLLATE = 'utf8mb4_unicode_520_ci';

CREATE TABLE \`article_urls\` (
  \`articleIntId\` INTEGER UNSIGNED NOT NULL,
  \`createdAt\` TIMESTAMP(3) NOT NULL DEFAULT NOW(3) COMMENT 'The date, fixed, of the document\\\\'s creation',
  \`updatedAt\` TIMESTAMP(3) NOT NULL DEFAULT NOW(3) ON UPDATE NOW(3) COMMENT 'The date of the document\\\\'s last update',
  \`path\` VARCHAR(255) NOT NULL,
  PRIMARY KEY (\`articleIntId\`),
  INDEX (\`articleIntId\`),
  FOREIGN KEY (\`articleIntId\`) REFERENCES \`articles\` (\`intId\`) ON UPDATE CASCADE ON DELETE RESTRICT
) ENGINE = 'InnoDB',
CHARSET = 'utf8mb4',
COLLATE = 'utf8mb4_unicode_520_ci';

CREATE TABLE \`article_url_metas\` (
  \`urlArticleIntId\` INTEGER UNSIGNED NOT NULL,
  \`createdAt\` TIMESTAMP(3) NOT NULL DEFAULT NOW(3) COMMENT 'The date, fixed, of the document\\\\'s creation',
  \`updatedAt\` TIMESTAMP(3) NOT NULL DEFAULT NOW(3) ON UPDATE NOW(3) COMMENT 'The date of the document\\\\'s last update',
  PRIMARY KEY (\`urlArticleIntId\`),
  INDEX (\`urlArticleIntId\`),
  FOREIGN KEY (\`urlArticleIntId\`) REFERENCES \`article_urls\` (\`articleIntId\`) ON UPDATE CASCADE ON DELETE RESTRICT
) ENGINE = 'InnoDB',
CHARSET = 'utf8mb4',
COLLATE = 'utf8mb4_unicode_520_ci';"
`;
